internal/paascharm/go.tmpl

```diff
@@ -0,0 +1,127 @@
+     HTTPSProxy *string `env:"HTTPS_PROXY"`
+     NoProxy []string `env:"NO_PROXY"`
+}
+
+{{ if .Integrations }}
+   type Integrations struct {
+   {{- range .Integrations }}
+       {{ .GoName }} {{ .GoName }}Integration {{ if .IsDatabase }}`envPrefix:"{{ .DatabasePrefix }}"`{{ end }}
+   {{- end }}
+   }
+{{ end }}
+
+
+{{ if .HasDatabaseIntegrations }}
+
+   {{- range .Integrations }}
+       {{ if .IsDatabase }}
+       type {{ .GoName }}Integration struct {
+            DatabaseIntegration
+       }
+       {{ end }}
+    {{ end }}
+
+    type DatabaseIntegration struct {
+         ConnectString string  `env:"DB_CONNECT_STRING"`
+         Scheme        string  `env:"DB_SCHEME"`
+         NetLoc        string  `env:"DB_NETLOC"`
+         Path          string  `env:"DB_PATH"`
+         Params        string  `env:"DB_PARAMS"`
+         Query         string  `env:"DB_QUERY"`
+         Fragment      string  `env:"DB_FRAGMENT"`
+         Username      *string `env:"DB_USERNAME"`
+         Hostname      *string `env:"DB_HOSTNAME"`
+         Password      *string `env:"DB_PASSWORD"`
+         Port          *int    `env:"DB_PORT"`
+         Name          *string `env:"DB_NAME"`
+    }
+
+    func (di DatabaseIntegration) IsActive() bool {
+         return di.ConnectString != ""
+    }
+
+{{ end }}
+
+{{ if .Integrations.s3 }}
+   {{- with .Integrations.s3 }}
+   type S3Integration struct {
+        AccessKey    string  `env:"APP_S3_ACCES1S_KEY"`
+        SecretKey    string  `env:"APP_S3_SECRET_KEY"`
+        Region       *string `env:"APP_S3_REGION"`
+        StorageClass *string `env:"APP_S3_STORAGE_CLASS"`
+        Bucket       string  `env:"APP_S3_BUCKET"`
+        Endpoint     *string `env:"APP_S3_ENDPOING"`
+        Path         *string `env:"APP_S3_PATH"`
+        ApiVersion   *string `env:"APP_S3_API_VERSION"`
+        UriStyle        *string `env:"APP_S3_URI_STYLE"`
+        AddressingStyle *string `env:"APP_S3_ADDRESSING_STYLE"`
+        Attributes *string `env:"APP_S3_ATTRIBUTES"`
+        TLSCAChain *string `env:"APP_S3_TLS_CA_CHAIN"`
+    }
+
+    func (s3 S3Integration) IsActive() bool {
+        if s3.AccessKey == "" {
+           return false
+        }
+        return true
+    }
+   {{ end }}
+{{ end }}
+
+
+{{ if .Integrations.saml }}
+   {{- with .Integrations.saml }}
+   type SAMLIntegration struct {
+        EntityID                string  `env:"APP_SAML_ENTITY_ID"`
+        MetadataURL             *string `env:"APP_SAML_METADATA_URL"`
+        SingleSignOnRedirectURL string  `env:"APP_SAML_SINGLE_SIGN_ON_REDIRECT_URL"`
+        SigningCertificate      string  `env:"APP_SAML_SIGNING_CERTIFICATE"`
+    }
+
+    func (si SAMLIntegration) IsActive() bool {
+        if si.EntityID == "" {
+            return false
+        }
+        return true
+    }
+
+   {{ end }}
+{{ end }}
+
+{{- range .Integrations }}
+    func (_ {{ .GoName }}Integration) IsOptional() bool {
+         return {{ .Optional }}
+    }
+{{ end }}
```

Comment:
Can't this be done in the previous `.Integrations` range ?

Reply:
This is the only place where the iteration is done on all the integrations without being under a "struct" (other times it is only the databases, of a specific database). Trying some changes I cannot get it much cleaner, any suggestion?